### PR TITLE
chore: use infra-checkers static analysis settings

### PR DESCRIPTION
**Pull request by sigilioso**

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -190,7 +196,3 @@ jobs:
       - uses: actions/checkout@v2
       - name: Codespell test
         uses: codespell-project/actions-codespell@master
-        with:
-          check_filenames: true
-          check_hidden: true
-          skip: ./.git,go.sum
```
Comment (gsanchezgavier) [inline .github/workflows/ci.yaml:196]: is it ok to remove this?  i didnt see any codespell configs in the infra-checkers repo.

File: Makefile [hunk 2]
```diff
@@ -12,13 +12,6 @@ BINARY_NAME ?= newrelic-infra-operator
 
 LD_FLAGS ?= "-extldflags '-static'"
 
-ifeq (, $(shell which golangci-lint))
-GOLANGCI_LINT ?= go run -modfile=tools/go.mod github.com/golangci/golangci-lint/cmd/golangci-lint
```
Comment (paologallinaharbur) [inline Makefile:16]: can get rid of https://github.com/newrelic/newrelic-infra-operator/blob/main/tools/tools.go and all horrible dependencies?

Comment (sigilioso) [reply Makefile:16]: Sure! I don't know why I missed it in this repo...

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (sigilioso) [reply .github/workflows/ci.yaml:196]: you are right, thank you!

Comment (sigilioso) [reply .github/workflows/ci.yaml]: Restored here: f73c416eef0f2de4b845b6722c3278960eb57fd6

File: Makefile (hunk 2 above)
Comment (sigilioso) [reply Makefile:16]: Done: 801dedbd12ae06f475cdc4cb0ab5367160eaa6b3

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -167,20 +167,26 @@ jobs:
...
+        uses: returntocorp/semgrep-action@v1
+        with:
+          auditOn: push
       - name: golangci-lint
-        uses: golangci/golangci-lint-action@v3.1.0
+        uses: golangci/golangci-lint-action@v3
+        continue-on-error: ${{ github.event_name != 'pull_request' }}
         with:
-          # Optional: show only new issues if it's a pull request. The default value is `false`.
           only-new-issues: true
```
Comment (paologallinaharbur) [inline .github/workflows/ci.yaml:189]: I feel like that bumping the version auto-enabled some additional linters and now we are taking more time. To solve the timeout issue in the ci we could try to add.
```
          args: --timeout 3m 
```

Possibly related to https://github.com/golangci/golangci-lint-action/issues/297

Comment (sigilioso) [reply .github/workflows/ci.yaml:189]: It seems to work, thanks!

**Review by paologallinaharbur — APPROVED** [review]

**Pull request merged**